third_party/nvfuser/csrc/sys_utils.cpp

```diff
@@ -0,0 +1,110 @@
+#include <c10/util/Exception.h>
+
+#if defined(__linux__)
+
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+namespace torch::jit::fuser::cuda::executor_utils {
+
+std::string disassembleBinary(const std::vector<char>& cubin) {
```

Comment:
Does it need to be done with the explicit `fork` and `exec` rather than just `popen`?

Reply:
I was originally using `popen`, but I found that the mode can only be `r` or `w`, but not `rw`. So I switched to fork + exec thinking that I can avoid using a temporary file with this approach. But unfortunately, I then found that, `nvdisasm` doesn't work with `/dev/stdin`, which means I have to create a temporary file anyway. I am just too tired of changing from `popen` to `fork` + `exec` and I don't want to change back... But on the other hand, I do feel that `fork` + `exec` is the cleanest way to manage temp file, because the creation and removal of tempfile are all in a single line of code.